src/main/java/mate/academy/dao/impl/MovieDaoImpl.java

```diff
@@ -0,0 +1,45 @@
+package mate.academy.dao.impl;
+
+import java.util.List;
+import javax.persistence.criteria.CriteriaQuery;
+import mate.academy.dao.MovieDao;
+import mate.academy.exceptions.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Movie;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        Session session = HibernateUtil.getSessionFactory().openSession();
+        try {
+            transaction = session.beginTransaction();
+            Long itemId = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(itemId);
```

Comment:
please see checklist: you don’t need to set id for your entity explicitly - it will be done by Hibernate :)